static/app/views/profiling/landing/profilingSlowestTransactionsPanel.tsx

```diff
@@ -0,0 +1,226 @@
+
+  return (
+    <PanelItem key={transaction.transaction}>
+      <Flex justify="space-between" gap={space(1)}>
+        <PlatformIcon platform={transactionProject?.platform} />
+        <Flex.Item grow={1}>
+          <Link
+            to={generateProfileSummaryRoute({
+              orgSlug: organization.slug,
+              projectSlug: transactionProject?.slug!,
+            })}
+          >
+            <TextTruncateOverflow>
+              {transaction.transaction as string}
+            </TextTruncateOverflow>
+          </Link>
+        </Flex.Item>
+
+        <PerformanceDuration nanoseconds={transaction['p95()'] as number} abbreviation />
+        <Button borderless size="zero" onClick={() => onOpen()}>
+          <IconChevron direction={open ? 'up' : 'down'} size="xs" />
+        </Button>
+      </Flex>
+      <PanelItemBody
+        style={{
+          height: open ? 160 : 0,
+        }}
+      >
+        {open && transactionProject && (
+          <PanelItemFunctionsMiniGrid
+            transaction={String(transaction.transaction)}
+            organization={organization}
+            project={transactionProject}
+          />
+        )}
+      </PanelItemBody>
+    </PanelItem>
+  );
+}
+
+interface PanelItemFunctionsMiniGridProps {
+  organization: Organization;
+  project: Project;
+  transaction: string;
+}
+
+function PanelItemFunctionsMiniGrid(props: PanelItemFunctionsMiniGridProps) {
+  const {transaction, project, organization} = props;
+  const {functionsQuery, functions} = useProfilingTransactionQuickSummary({
+    transaction,
+    project,
+    referrer: 'api.profiling.landing-slowest-transaction-panel',
+    skipLatestProfile: true,
+    skipSlowestProfile: true,
+  });
+
+  if (functionsQuery.type === 'loading') {
+    return <FunctionsMiniGridLoading />;
+  }
+
+  if (functionsQuery.type === 'resolved' && functions && functions.length === 0) {
+    return <FunctionsMiniGridEmptyState />;
+  }
+
+  return (
+    <PanelItemBodyInner>
+      <FunctionsMiniGrid
+        functions={functions}
+        organization={organization}
+        project={project}
+      />
+    </PanelItemBodyInner>
+  );
+}
+
+const FlexPanel = styled(Panel)`
+  display: flex;
+  flex-direction: column;
+`;
+
+const PanelHeading = styled('span')`
+  font-size: ${p => p.theme.text.cardTitle.fontSize};
+  font-weight: ${p => p.theme.text.cardTitle.fontWeight};
+  line-height: ${p => p.theme.text.cardTitle.lineHeight};
+`;
+
+const PanelSubheading = styled('span')`
+  color: ${p => p.theme.subText};
+`;
+
+const PanelItem = styled('div')`
+  padding: ${space(1)} ${space(1.5)};
+  border-top: 1px solid ${p => p.theme.border};
+`;
+const PanelItemBody = styled('div')`
+  transition: height 0.1s ease;
+  width: 100%;
+  overflow: hidden;
+`;
+
+// TODO: simple layout stuff like this should come from a primitive component and we should really stop this `styled` nonsense
+const PanelItemBodyInner = styled('div')`
+  padding-top: ${space(1.5)};
+`;
```

Comment:
Ideally we have a primitive component for spacing/layout... other ui libs typically call it `<Box />` 

There are alot of these small 1-off `styled` components that i think we can do away with and reserve the usage of `styled` to specialized use cases

`PanelHeading` and `PanelSubheading` are also examples of this. A simple `<Heading />` and `<Text />` component with a set of `variants` and `sizes` should handle 80-90% of use cases.